utask.go

```diff
@@ -157,39 +163,99 @@ type DatabaseConfig struct {
 
 func (c *Cfg) buildLimits() {
 	c.resourceSemaphores = make(map[string]*semaphore.Weighted)
+	c.deadResources = make(map[string]struct{})
 
 	for k, v := range c.ResourceLimits {
-		c.resourceSemaphores[k] = semaphore.NewWeighted(int64(v))
+		if v <= 0 {
```

Comment:
Would it make sense to have 0 mean "nothing allowed" and negative values mean "disregard that config, return to default behavior a.k.a not care about limit" ?

Right now if I want to disable a limit, I have to take it out of the cfg. This would allow me to flip back and forth.

Reply:
We could have that.
I just didn't wanted to change the current behaviour too much, as the pull request is already big.
We allow the negative value right now, but that creates semaphores that are blocking infinitely, until the context is cancelled (current context being `context.Background()`) ... https://github.com/golang/sync/blob/master/semaphore/semaphore.go#L49